### PR TITLE
Join 2 articles public form views into one for sake of DRY

### PR DESCRIPTION
**Pull request by ramusus**
Solves #336

### CODE REVIEW

**Comment by andresmanz** [comment]
In my opinion it is extremely helpful to have best practices in demo code. Is there a reason this hasn't been merged yet?

**Comment by codydaig** [comment]
@ramusus Would you mind updating this PR against the latest master branch? I'd really like to see this change myself. Unless there's any objections of moving to a dry structure? @lirantal @ilanbiala @mleanos @trainerbill @rhutchison

File: public/modules/articles/controllers/articles.client.controller.js [hunk 1]
```diff
@@ -4,19 +4,21 @@ angular.module('articles').controller('ArticlesController', ['$scope', '$statePa
...
+			var success = function(response) {
 				$location.path('articles/' + response._id);
-
-				$scope.title = '';
-				$scope.content = '';
-			}, function(errorResponse) {
+				$scope.article.title = '';
+				$scope.article.content = '';
+			}
+			var error = function(errorResponse) {
```
Comment (ilanbiala) [inline public/modules/articles/controllers/articles.client.controller.js]: What is this doing now?

**Comment by ilanbiala** [comment]
@ramusus this looks fine to me, albeit some changes that need to be made such as removing `data-` and fixing the indentation.

**Comment by ilanbiala** [comment]
@ramusus would you be able to tidy up this PR so it can be merged?

**Comment by ramusus** [comment]
Sorry, unfortunatly I'm very busy now

**Comment by mleanos** [comment]
@ilanbiala @codydaig 

This looks like a good implementation. Since @ramusus doesn't have time for this, I'll take it over. This is related to #807 & #874 Once the latter is decided on, then I will work on this implementation.

**Comment by ilanbiala** [comment]
@mleanos why do you need #874 to work on this?

**Comment by mleanos** [comment]
@ilanbiala #874 is a refactor of the Article module. However, since that PR is set for 0.5.0 then this can probably move forward with consideration for both #807 and #874

**Comment by codydaig** [comment]
@ilanbiala Can this be moved to 0.5.0?

**Comment by ilanbiala** [comment]
mleanos do you need #874 to work on this?

**Comment by mleanos** [comment]
@ilanbiala I'm not sure. I need to take another look at this implementation. I'd rather wait until the dust settles with #874 since it's a huge refactor, and this implementation might not be relevant once we implement John Papa's styleguide; I'm not exactly sure though.

Also, #807 will introduce a change in how we're allowing users to edit/create Articles, it may make sense to hold off on this; or just address these concerns in #807

**Comment by rhutchison** [comment]
@ramusus I agree with the direction, but not completely with the implementation.  The controller should call $save and the service should work out whether it's an insert or update.

@codydaig @mleanos

**Comment by mleanos** [comment]
@rhutchison I agree. I would like to reintroduce this with #807. Also, would prefer to add the angular styleguide with it, since the article refactor PR has stalled for so long. We need to get moving on the styleguide, so we can have it out with the 0.5.0 release.

**Pull request closed without merging**